src/auto-evo/AutoEvoExploringTool.cs

```diff
@@ -0,0 +1,640 @@
+        autoEvoConfiguration.ProtectNewCellsFromSpeciesCap = protectNewCellsFromSpeciesCapCheckBox.Pressed;
+        autoEvoConfiguration.RefundMigrationsInExtinctions = refundMigrationsInExtinctionsCheckBox.Pressed;
+        autoEvoConfiguration.StrictNicheCompetition = strictNicheCompetitionCheckBox.Pressed;
+        autoEvoConfiguration.SpeciesSplitByMutationThresholdPopulationAmount =
+            (int)speciesSplitByMutationThresholdPopulationAmountSpinBox.Value;
+        autoEvoConfiguration.SpeciesSplitByMutationThresholdPopulationFraction =
+            (float)speciesSplitByMutationThresholdPopulationFractionSpinBox.Value;
+        autoEvoConfiguration.UseBiodiversityForceSplit = useBiodiversityForceSplitCheckBox.Pressed;
+    }
+
+    /// <summary>
+    ///   Run a new generation or finish the current generation async
+    /// </summary>
+    private void OnFinishOneGenerationButtonPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        if (autoEvoRun?.Aborted != false || autoEvoRun.Finished)
+        {
+            // If the previous one has finished / failed
+            autoEvoRun = new AutoEvoRun(gameProperties.GameWorld, autoEvoConfiguration) { FullSpeed = true };
+            autoEvoRun.Start();
+        }
+        else
+        {
+            // If the previous one is paused
+            autoEvoRun.FullSpeed = true;
+            autoEvoRun.Continue();
+        }
+
+        // Disable these buttons
+        finishOneGenerationButton.Disabled = true;
+        runOneStepButton.Disabled = true;
+        abortButton.Disabled = false;
+    }
+
+    /// <summary>
+    ///   Apply auto-evo effects and set up related controls
+    /// </summary>
+    private void ApplyAutoEvoRun()
+    {
+        // Apply the results
+        autoEvoRun!.ApplyAllEffects(true);
+
+        // Add run results
+        RunResults results = autoEvoRun.Results!;
+        evolutionaryTree.UpdateEvolutionaryTreeWithRunResults(results, ++currentGeneration);
+        runResultsList.Add(results.MakeSummary(gameProperties.GameWorld.Map, true));
+        speciesHistoryList.Add(
+            gameProperties.GameWorld.Species.ToDictionary(pair => pair.Key, pair => (Species)pair.Value.Clone()));
+
+        // Add check box to history container
+        historyListMenu.AddItem(currentGeneration.ToString(), false, Colors.White);
+        historyListMenu.CreateElements();
+
+        currentGenerationLabel.Text = currentGeneration.ToString();
+    }
+
+    /// <summary>
+    ///   Run one step sync
+    /// </summary>
+    private void OnRunOneStepButtonPressed()
+    {
+        GUICommon.Instance.PlayButtonPressSound();
+
+        if (autoEvoRun?.Aborted != false || autoEvoRun.Finished)
+        {
+            autoEvoRun = new AutoEvoRun(gameProperties.GameWorld, autoEvoConfiguration);
+        }
+
+        // To avoid concurrent steps
+        autoEvoRun.FullSpeed = false;
```

Comment:
I'm not convinced that setting full speed to false here actually prevents concurrent steps. Shouldn't this wait for currently executing auto-evo step tasks to stop? Or can this method only be called when auto-evo hasn't been running in the background previously?

Reply:
This button is enabled only when auto-evo is paused or not running.